### PR TITLE
First attempt at fixing issue 2454: making consider_constant() functiona...

### PR DESCRIPTION
**Pull request by julianser**
...lity equivalent to consider_constant arguement in T.grad().

### CODE REVIEW

**Comment by lamblin** [comment]
Please link to the issue in a comment: #2454.

**Comment by carriepl** [comment]
This looks good but (as we discussed in person, I'm just putting this here to have a trace of that discussion) this also requires some tests.

**Comment by carriepl** [comment]
This will need to be update because some automated tests are failing. You can come see me if you don't know how to check which ones.

File: theano/gradient.py [hunk 1]
```diff
@@ -1878,6 +1878,7 @@ def _is_zero(x):
     return 'yes'
 
 
+
```
Comment (carriepl) [inline theano/gradient.py]: Remove this line

File: theano/tests/test_gradient.py [hunk 2]
```diff
@@ -603,7 +603,6 @@ def test_subgraph_grad():
     for true_grad, pgrad in zip(true_grads, pgrads):
         assert(np.sum(np.abs(true_grad - pgrad)) < 0.00001)
 
-
```
Comment (carriepl) [inline theano/tests/test_gradient.py]: The line removed here should be added back; convention in python is two empty lines before a class declaration.

File: theano/tests/test_gradient.py [hunk 3]
```diff
@@ -642,7 +641,110 @@ def test_grad(self):
...
+            g = gradient.grad(expr.sum(), x)
+            # gradient according to theano
+            f = theano.function([x], g, on_unused_input='ignore')
+            # desired gradient
+            f2 = theano.function([x], expr_grad, on_unused_input='ignore')
+
+            assert np.allclose(f(a), f2(a))
+
+def grad_wrt_expression(exp, grad_wrt_variable):
+    return gradient.grad(exp, grad_wrt_variable)
```
Comment (carriepl) [inline theano/tests/test_gradient.py]: What is the point of this function?

**Comment by carriepl** [comment]
You also have a bunch or lines with trailing space characters which should be removed. Other than that, it seems pretty good.

File: theano/tests/test_gradient.py (hunk 3 above)
Comment (julianser) [inline theano/tests/test_gradient.py]: Nothing actually, it was a function I used earlier but now got rid of it... I'll remove it :)

**Comment by carriepl** [comment]
Tests are failing

**Comment by carriepl** [comment]
There was a timeout for one set of unit tests so I restarded it. I'll continue the review once it's done.

**Comment by carriepl** [comment]
Looks good. Thanks.

**Pull request merged**